File: app/src/main/java/com/mgaetan89/showsrage/presenter/ShowPresenter.java
package com.mgaetan89.showsrage.presenter;

import android.support.annotation.Nullable;
import android.support.annotation.StringRes;

import com.mgaetan89.showsrage.R;
import com.mgaetan89.showsrage.model.Indexer;
import com.mgaetan89.showsrage.model.Show;
import com.mgaetan89.showsrage.network.SickRageApi;

public class ShowPresenter {
	@Nullable
	private Show show;

	public ShowPresenter(@Nullable Show show) {
		this.show = show;
	}

	public int getDownloaded() {
		if (this.show == null) {
			return 0;
		}

		return this.show.getDownloaded();
	}

	public int getEpisodesCount() {
		if (this.show == null) {
			return 0;
		}

		return this.show.getEpisodesCount();
	}

	public String getNetwork() {
		if (this.show == null) {
			return "";
		}

		return this.show.getNetwork();
	}

	@StringRes
	public int getPauseActiveStatusRes() {
		if (this.show == null) {
			return 0;
		}

		if (this.show.getPaused() == 0) {
			return R.string.active;
		}

		return R.string.paused;
	}

	public String getPosterUrl() {
		if (this.show == null) {
			return "";
		}

		return SickRageApi.getInstance().getPosterUrl(this.show.getTvDbId(), Indexer.TVDB);
	}

	public String getQuality() {
		if (this.show == null) {
			return "";
		}

		return this.show.getQuality();
	}

	public String getShowName() {
		if (this.show == null) {
			return "";
		}

		return this.show.getShowName();
	}

	public int getSnatched() {
		if (this.show == null) {
			return 0;
		}

		return this.show.getSnatched();
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/presenter/ShowPresenter.kt
package com.mgaetan89.showsrage.presenter

import com.mgaetan89.showsrage.extension.getShowStat
import com.mgaetan89.showsrage.model.Indexer
import com.mgaetan89.showsrage.model.RealmShowStat
import com.mgaetan89.showsrage.model.Show
import com.mgaetan89.showsrage.network.SickRageApi
import io.realm.Realm

class ShowPresenter(val show: Show?) {
	fun getBannerUrl() = if (this.isShowValid()) SickRageApi.instance.getBannerUrl(this.show!!.tvDbId, Indexer.TVDB) else ""

	fun getNetwork() = if (this.isShowValid()) this.show!!.network else ""

	fun getPosterUrl() = if (this.isShowValid()) SickRageApi.instance.getPosterUrl(this.show!!.tvDbId, Indexer.TVDB) else ""

	fun getQuality() = if (this.isShowValid()) this.show!!.quality else ""

	fun getShowName() = if (this.isShowValid()) this.show!!.showName else ""

	fun getShowStat(): RealmShowStat? {
		if (!this.isShowValid()) {
			return null
		}

		if (this.show!!.stat == null) {
			this.show.stat = Realm.getDefaultInstance().use {
				it.getShowStat(this.show.indexerId)
			}
		}

		return this.show.stat
	}

	fun isPaused() = if (this.isShowValid()) this.show!!.paused == 1 else false

	internal fun isShowValid() = this.show != null && this.show.isValid
}
